### ISSUE TITLE
[Feature Request] Add `replace` API similar to dataclasses.replace

### ISSUE THREAD

**Issue by zhiruiluo** [lebrice/SimpleParsing#198]
**Is your feature request related to a problem? Please describe.**
The replace function of the dataclasses module has the signature of [Dataclasses.replace(obj, /, **changes)](https://docs.python.org/3/library/dataclasses.html#dataclasses.replace):
- it creates a new object of the same type as obj, replacing fields with values from changes.
- If obj is not a Data Class, raises TypeError. If values in changes do not specify fields, raises TypeError.

However, the Dataclass.replace doesn't work with nested dataclasses, subgroups, and other features in simple-parsing. To solve this, the simple_parsing.replace should be supplemented as an extension to dataclasses.replace.

**Describe the solution you'd like**
We can use the `parse` API to implement this feature.

**Describe alternatives you've considered**

**Additional context**
With `simple_parsing.replace`, we can make this example work.
```python
import simple_parsing as sp

@dataclass
class InnerClass:
    arg1: int = 0
    arg2: str = "foo"


@dataclass(frozen=True)
class OuterClass:
    outarg: int = 1
    nested: InnerClass = InnerClass()


changes_1 = {"outarg": 2, "nested.arg1": 1, "nested.arg2": "bar"}
changes_2 = {"outarg": 2, "nested": {"arg1": 1, "arg2": "bar"}}
c = OuterClass()
c1 = sp.replace(c, changes_1)
c2 = sp.replace(c, changes_2)
assert c1 == c2
assert c1.outarg == 2
assert c1.nested.arg1 == 1
assert c1.nested.arg2 == "bar"
```

**Comment by zhiruiluo** [comment]
I have proposed a PR #197 to this feature request.

**Comment by zhiruiluo** [comment]
`simple_parsing.replace` has been merged in #212
